sol/private/toolchains_repo.bzl

```diff
@@ -79,14 +80,16 @@ load(":defs.bzl", "resolved_toolchain")
 resolved_toolchain(name = "resolved_toolchain", visibility = ["//visibility:public"])
 
 """
+    version_constraint = "@aspect_rules_sol//sol/private:solc_version_{}".format(normalize_version_string(repository_ctx.attr.version))
 
     for [platform, meta] in PLATFORMS.items():
         build_content += """
 # Declare a toolchain Bazel will select for running the tool in an action
 # on the execution platform.
 toolchain(
     name = "{platform}_toolchain",
-    exec_compatible_with = {compatible_with},
+    target_compatible_with = {compatible_with},
```

Comment:
@thesayyn this seems undesirable, but I guess transitions cannot do anything about the effective exec platform, we have to pretend to target a solc runtime of a particular version.